crates/ruma-events/src/room/server_acl.rs

```diff
@@ -45,10 +47,34 @@ impl RoomServerAclEventContent {
     pub fn new(allow_ip_literals: bool, allow: Vec<String>, deny: Vec<String>) -> Self {
         Self { allow_ip_literals, allow, deny }
     }
+
+    /// Returns true if and only if the server is allowed by the ACL rules.
+    pub fn is_allowed(&self, server_name: &ServerName) -> bool {
+        if !self.allow_ip_literals && server_name.is_ip_literal() {
+            return false;
+        }
+
+        let host = server_name.host();
+
+        for deny in &self.deny {
+            if WildMatch::new(deny).matches(host) {
+                return false;
+            }
+        }
+
+        for allow in &self.allow {
+            if WildMatch::new(allow).matches(host) {
+                return true;
+            }
+        }
+
+        false
```

Comment:
This can be condensed to just `self.deny.iter().all(/* not match */) && self.allow.any(/* match */)`.